sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1624,6 +1740,8 @@ class HistGradientBoostingClassifier(ClassifierMixin, BaseHistGradientBoosting):
           features.
         - str array-like: names of categorical features (assuming the training
           data has feature names).
+        - `"pandas"`: Pandas categorical dtypes are considered categorical. The
+          input must be a pandas DataFrame to use this feature.
```

Comment:
For the better UX, I am in favor of making it the default option. I have two concerns with making it the default option: 

1. Categorical features backed by numeric data are converted to a float ndarray:

```python
from sklearn.utils.validation import check_array
import pandas as pd

df = pd.DataFrame({"a": [1, 2, 3], "b": pd.Series([3.0, 1., 9.], dtype="category")})
X = check_array(df)
print(X.dtype)
# float64
```

On `main`, the categorical feature was treated as a numerical feature and will be a categorical feature with this PR. This is a behavior change.

2. On `main`, the categorical features with cardinality greater than `max_bins` work, but will fail with this PR because the feature is automatically considered categorical. Concretely, the following works on main:

```python
import pandas as pd
from sklearn.datasets import make_regression
from sklearn.ensemble import HistGradientBoostingClassifier

rng = np.random.RandomState(42)
X_cat = rng.randint(0, 400, size=1_000)
y = X_cat % 2
df = pd.DataFrame({"a": pd.Series(X_cat, dtype="category")})

hist = HistGradientBoostingClassifier(random_state=0)
hist.fit(df, y)
```

Reply:
One could argue that the current behavior on main (treating explicitly categorical dtyped features as numerical) is a (minor) bug.

Reply:
Maybe we can be conservative an issue a futurewarning for now when the input has some categorical dtype and says that in the future `by_dtype` will be the default and that in the mean time it's possible to set `categorical_features="by_dtype"` or `categorical_features="None"` explicitly to silence the warning.